File: 49-group-anagrams/group-anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    const map = {}

    for (const str of strs) {
        const sortedStr: string = str.split('').sort().join('')

        if (!map[sortedStr]) {
            map[sortedStr] = [str]
        } else {
            map[sortedStr].push(str)
        }
    }

    return Object.values(map)
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = defaultdict(list)
        base = ord('a')

        for s in strs:
            counts = [0] * 26

            for ch in s:
                counts[ord(ch) - base] += 1
            
            key = '#'.join(map(str, counts))
            groups[key].append(s)

        return list(groups.values())